pkg/crc/config/viper_config.go

```diff
@@ -33,7 +34,11 @@ func NewViperStorage(configFile, envPrefix string) (*ViperStorage, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error reading configuration file '%s': %v", configFile, err)
 	}
-	v.WatchConfig()
+	// only watch config file if not running in a terminal/interactively
+	if !terminal.IsTerminal(int(os.Stdout.Fd())) {
```

Comment:
This is hardcoding some crc-specific knowledge in a package which should be generic. I'd prefer that you add a "WatchConfig()" method to the viper storage that the daemon calls explicitly.